src/routing_table/network_tests.rs

```diff
@@ -17,297 +17,244 @@
 
 #![cfg(test)]
 
+use maidsafe_utilities::SeededRng;
+use rand::Rng;
+use std::cmp;
+use std::collections::{BTreeSet, HashMap, HashSet};
+use std::collections::hash_map::Entry;
+use std::fmt::{self, Binary, Debug, Formatter};
+use super::{Destination, RoutingTable};
+use super::xorable::Xorable;
 
-// use maidsafe_utilities::SeededRng;
-// use rand::Rng;
-// use std::cmp;
-// use std::collections::{HashMap, HashSet};
-// use std::collections::hash_map::Entry;
-// use std::fmt::{self, Binary, Debug, Formatter};
-// use super::routing_table::{Destination, RoutingTable};
-// use super::xorable::Xorable;
+const GROUP_SIZE: usize = 8;
 
-// const GROUP_SIZE: usize = 8;
+#[derive(Clone, Eq, PartialEq)]
+struct Contact(u64);
 
-// #[derive(Clone, Eq, PartialEq)]
-// struct Contact(u64);
+/// A simulated network, consisting of a set of "nodes" (routing tables) and a random number
+/// generator.
+#[derive(Default)]
+struct Network {
+    rng: SeededRng,
+    nodes: HashMap<u64, RoutingTable<u64>>,
+}
 
-// /// A simulated network, consisting of a set of "nodes" (routing tables) and a random number
-// /// generator.
-// #[derive(Default)]
-// struct Network {
-//     rng: SeededRng,
-//     nodes: HashMap<u64, RoutingTable<u64>>,
-// }
+impl Network {
+    /// Creates a new empty network with a seeded random number generator.
+    fn new(optional_seed: Option<[u32; 4]>) -> Network {
+        Network {
+            rng: match optional_seed {
+                Some(seed) => SeededRng::from_seed(seed),
+                None => SeededRng::new(),
+            },
```

Comment:
This could be shortened to `optional_seed.map_or_else(SeededRng::new, SeededRng::from_seed)`.